src/compiler/checker.ts

```diff
@@ -2801,8 +2812,8 @@ namespace ts {
             const namespaceMeaning = SymbolFlags.Namespace | (isInJSFile(name) ? meaning & SymbolFlags.Value : 0);
             let symbol: Symbol | undefined;
             if (name.kind === SyntaxKind.Identifier) {
-                const message = meaning === namespaceMeaning ? Diagnostics.Cannot_find_namespace_0 : getCannotFindNameDiagnosticForName(getFirstIdentifier(name));
-                const symbolFromJSPrototype = isInJSFile(name) ? resolveEntityNameFromAssignmentDeclaration(name, meaning) : undefined;
+                const message = meaning === namespaceMeaning || nodeIsSynthesized(name) ? Diagnostics.Cannot_find_namespace_0 : getCannotFindNameDiagnosticForName(getFirstIdentifier(name));
+                const symbolFromJSPrototype = isInJSFile(name) && !nodeIsSynthesized(name) ? resolveEntityNameFromAssignmentDeclaration(name, meaning) : undefined;

```

Comment:
I just hit this case when doing `resolveEntityName` on an identifier, we report `Cannot find namespace` on an identifier that isn't a namespace (which is why the `meaning === namespaceMeaning` check existied in the first place). We should only be reporting `Cannot find namespace` when the id is part of a qualified name. What was the reason to report `Cannot find namespace` when the node is synthetic? Is it because `getCannotFindNameDiagnosticForName` uses `.parent`?

Reply:
It's been awhile, but that's probably a good guess.